src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php

```diff
@@ -329,6 +332,61 @@ public function testContextIsPopulatedWithFormBeingValidatedUsingGroupSequence()
 
         $this->assertCount(0, $violations);
     }
+
+    public function testSubmitFormChoiceInvalid()
+    {
+        $form = $this->formFactory->create(DateType::class, null, [
+            'widget' => 'choice',
+            'years' => [2021],
+        ]);
+
+        $form->submit([
+            'day' => '13',
+            'month' => '13',
+            'year' => '2020',
+        ]);
+
+        $this->assertTrue($form->isSubmitted());
+        $this->assertFalse($form->isValid());
+        $this->assertCount(2, $form->getErrors());
+        $this->assertSame('This value is not valid.', $form->getErrors()[0]->getMessage());
+        $this->assertSame('This value is not valid.', $form->getErrors()[1]->getMessage());
```

Comment:
I wonder why this error is added twice.

Reply:
Once for the month and once for the year

Reply:
The removed `continue` is responsible for it. Maybe we should keep it?

Reply:
Hah, true. I didn't notice the invalid month. Was focused on the year only due to my initial test. 😄

Reply:
Should the cause of the error be tested as well then? To ensure one is really caused by the month and one by the year.

Reply:
3.4.40 also adds two errors in this case so it should be fine to keep that behavior.

Reply:
> Should the cause of the error be tested as well then? To ensure one is really caused by the month and one by the year.

Yes, totally makes sense. I did that.